pkg/apis/garden/v1beta1/helper/helpers.go

```diff
@@ -278,13 +322,17 @@ func DetermineMachineImage(cloudProfile gardenv1beta1.CloudProfile, name string)
 		return false, nil, fmt.Errorf("unknown cloud provider %s", cloudProvider)
 	}
```

Comment:
Why not using `GetMachineImagesFromCloudProfile` here instead of having the switch again?